py/sentry_relay/consts.py

```diff
@@ -1,25 +1,98 @@
-__all__ = ["SPAN_STATUS_CODE_TO_NAME", "SPAN_STATUS_NAME_TO_CODE"]
-
-
-SPAN_STATUS_CODE_TO_NAME = {
-    0: "ok",
-    1: "cancelled",
-    2: "unknown_error",
-    3: "invalid_argument",
-    4: "deadline_exceeded",
-    5: "not_found",
-    6: "already_exists",
-    7: "permission_denied",
-    8: "resource_exhausted",
-    9: "failed_precondition",
-    10: "aborted",
-    11: "out_of_range",
-    12: "unimplemented",
-    13: "internal_error",
-    14: "unavailable",
-    15: "data_loss",
-    16: "unauthenticated",
-}
-
-SPAN_STATUS_NAME_TO_CODE = dict((v, k) for k, v in SPAN_STATUS_CODE_TO_NAME.items())
-SPAN_STATUS_NAME_TO_CODE["unknown"] = SPAN_STATUS_NAME_TO_CODE["unknown_error"]
+from enum import IntEnum
+
+from sentry_relay._lowlevel import lib
+from sentry_relay.utils import decode_str, encode_str
+
+
+__all__ = ["DataCategory", "SPAN_STATUS_CODE_TO_NAME", "SPAN_STATUS_NAME_TO_CODE"]
+
+
+class BaseDataCategory(object):
+    @classmethod
+    def parse(cls, name):
+        """
+        Parses a `DataCategory` from its API name.
+        """
+        category = cls(lib.relay_data_category_parse(encode_str(name or "")))
+        if category == DataCategory.UNKNOWN:
+            return None  # Unknown is a Rust-only value, replace with None
+        return category
+
+    @classmethod
+    def from_event_type(cls, event_type):
+        """
+        Parses a `DataCategory` from an event type.
+        """
+        s = encode_str(event_type or "")
+        return cls(lib.relay_data_category_from_event_type(s))
+
+    @classmethod
+    def event_categories(cls):
+        """
+        Returns categories that count as events, including transactions.
+        """
+        return frozenset(
+            DataCategory.DEFAULT,
+            DataCategory.ERROR,
+            DataCategory.TRANSACTION,
+            DataCategory.SECURITY,
+        )
+
+    @classmethod
+    def error_categories(cls):
+        """
+        Returns categories that count as traditional error tracking events.
+        """
+        return frozenset(
+            DataCategory.DEFAULT, DataCategory.ERROR, DataCategory.SECURITY
```

Comment:
Note that this changes compared to Sentry, where `SECURITY` was not included:

https://github.com/getsentry/sentry/blob/9f08305e09866c8bd6d0c24f5b0aabdd7dd6c59c/src/sentry/constants.py#L532-L533